src/CloudSecrets/SecretsAnalyzer.cs

```diff
@@ -137,8 +137,9 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
 
         private bool IsRuleActive(string ruleKey)
         {
+            var rulesSettings = ruleSettingsProvider.Get();
```

Comment:
the ruleSettingsProvider already deals with checking if we are in connected mode and if the settings file exists so we can just replace the usersettingsProvider here.

The only issue is that the RuleSettingProvider will just say "unable to load connected mode settings" if we are in SonarQube < v9.9 instead of giving a more precise log message.

Reply:
Yes, in the ruleSettingsProvider it returns "userSettingsProvider.UserSettings.RulesSettings" if it cannot find the connected mode